src/views/internal/student/class/score/ClassScore.vue

```diff
@@ -89,13 +92,22 @@ export default {
             this.isLoading = status;
         },
         scoreGreat() {
-            return this.points === 10 ? true : false
+            return this.points === 9 ? true : false
         },
-        scoreRegular() {
+        scoreRegularOne() {
+            return this.points === 8 ? true : false
+        },
+        scoreRegularTwo() {
+            return this.points === 7 ? true : false
+        },
+        scoreRegularThree() {
+            return this.points === 6 ? true : false
+        },
+        scoreRegularFour() {
             return this.points === 5 ? true : false
         },
         scoreBad() {
-            return this.points === 3 ? true : false
+            return this.points > 0 && this.points < 5 ? true : false
         },
```

Comment:
Não consegui entender muito bem o motivo de ter tantos métodos especializados, acredito que não está sendo uma boa prática manter todos esses métodos.

Reply:
Criei os métodos pra servir como v-if da tag <progress>. Essa tag precisa de um value para preencher o conteúdo da barra. Contudo, não consegui passar esse valor dinamicamente, então optei por fazer dessa forma. Se tiver outra solução me avisa.

Reply:
Bom, estive pensando a respeito do problema e acabei chegando no seguinte código:

Primeiro, podemos definir no state as constantes de pontuação máxima e mínima para nos auxiliar nas verificações.

``` javascript
data() {
        return {
            MAX_SCORE: 9, // Pontuação máxima
            MIN_SCORE: 3, // Pontuação mínima
            points: null,
            time: null,
            uid: null,
            isLoading: true
        }
    }
```
Certo, agora podemos olhar para o nosso bloco template e simplificar os múltiplos blocos de progress. Aqui eu passei o parâmetro max e a pontuação de maneira dinâmica (a partir do que foi definido no state).
Note que também crei duas condições: isGreatScore e isRegularScore que estão no bloco computed. No caso de ser uma boa pontuação será renderizado um progress verde, em caso de um score regular será amarelo e no último (else) caso será vermelho.

``` html
<p style="font-size: 12px; margin-left: 5px">Progresso: ({{points}}/{{MAX_SCORE }})</p>
<progress v-if="isGreatScore" class="nes-progress is-success" :value="points" :max="MAX_SCORE"></progress>
<progress v-if="isRegularScore" class="nes-progress is-warning" :value="points" :max="MAX_SCORE"></progress>
<progress v-else class="nes-progress is-error" :value="points" :max="MAX_SCORE"></progress>
```

Aqui podemos ver os métodos que comentei acima. Basicamente o método isGreatScore retornará true apenas se a pontuação do usuário for máxima e o método isRegularScore retornará true apenas se a pontuação do usuário for entre a pontuação mínima e máxima (4, 5, 6, 7 ou 8).

```
    computed: {
        canIShowClass: function () {
            return !this.isLoading;
        },
        isGreatScore: function () {
            return this.points === this.MAX_SCORE;
        },
        isRegularScore: function () {
            return ((this.points > this.MIN_SCORE) && (this.points < this.MAX_SCORE))
        }
    }
```

Você acredita que isso facilita a solução ou complicou mais? hahaha